### DISCUSSION TITLE
Participants decision to withdraw from research / delete data set

### DISCUSSION THREAD

**Discussion by InaQuadt** [jspsych/jsPsych#1363] [Q&A] [answered]
Hi everyone,

I have managed to put together a Simon task using www.cognition.run.
So far, I am happy with it, but I wonder how I would go about deleting participants' data if they wish to withdraw from the study after I have collected the data. 
At the beginning of the experiment, I say something like "you have the right to withdraw at any time during and after the experiment."
If a participant emails me because he wants his data removed, how can I identify the right data set?
Is there a way of telling the participant "Here is your participation number, write it down somewhere in case you want your data removed from the study"?  

Best wishes,
Ina.

**Comment by becky-gilbert** [comment, marked as answer]
Hi Ina, you're right that in order to give participants this option, you'll need to give them some way of identifying their own data set after the experiment is over. I've seen other researchers do what you suggest - give participants a unique number that they can write down and give to you if they decide to withdraw later. 

Instead of providing a number, another option is to provide participants with a link that they can use to withdraw and a delete their own data. This link would go to a separate 'withdraw' study, which does nothing but records their unique ID so that you can delete it from the real study data. When you provide participants with this link during the real study, you can append the participant's unique ID to the end of this link, e.g. "www.example.com?id=123". The 'withdraw' study would automatically save the 'id' [URL variable to the data](https://www.jspsych.org/core_library/jspsych-data/#jspsychdataurlvariables), and show a message like "You have successfully withdrawn from <study name> and your results will be deleted." 

Since this question isn't specific to jsPsych, but relates to online experiments more generally, you could also try posting your question on this [online experiments forum](https://groups.google.com/forum/#!forum/online-experiments) to see if other people have suggestions. And if you need any help with implementation in jsPsych, then feel free to follow up here.

**Reply by InaQuadt** [reply]
Hi Becky,

Thank you for pointing me in the right direction if I have further questions regarding online experiments in general! 
For now, I will just give participants a number which they will have to write down. 

```javascript
var ID = {
	type: 'html-button-response',
  choices: ['ok'],
	prompt: '<p>This is your individual Participant ID.</p>' +
	        '<p style="color:red;">Please write it down in case you want to withdraw your data from the study at any point.</p>',
	stimulus: function(){
    return jsPsych.randomization.randomID(15);
	}
}
```

Thanks again,
Ina.
